src/main/java/org/fogbowcloud/manager/core/datastore/DatabaseManager.java

```diff
@@ -0,0 +1,100 @@
+package org.fogbowcloud.manager.core.datastore;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.datastore.orderstorage.*;
+import org.fogbowcloud.manager.core.exceptions.FatalErrorException;
+import org.fogbowcloud.manager.core.models.linkedlists.SynchronizedDoublyLinkedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+
+import java.sql.SQLException;
+
+public class DatabaseManager implements StableStorage {
+
+    private static final Logger LOGGER = Logger.getLogger(DatabaseManager.class);
+
+    private static DatabaseManager instance;
+
+    private ComputeOrderStorage computeOrderStorage;
+    private NetworkOrderStorage networkOrderStorage;
+    private VolumeOrderStorage volumeOrderStorage;
+    private AttachmentOrderStorage attachmentOrderStorage;
+    private OrderTimestampStorage orderTimestampStorage;
+
+    private DatabaseManager() throws SQLException {
+        this.computeOrderStorage = new ComputeOrderStorage();
+        this.networkOrderStorage = new NetworkOrderStorage();
+        this.volumeOrderStorage = new VolumeOrderStorage();
+        this.attachmentOrderStorage = new AttachmentOrderStorage();
+        this.orderTimestampStorage = new OrderTimestampStorage();
+    }
+
+    public static DatabaseManager getInstance() {
+        if (instance == null) {
+            try {
+                instance = new DatabaseManager();
+            } catch (SQLException e) {
+                LOGGER.error("Error instantiating database manager", e);
+                throw new FatalErrorException(e.getMessage());
+            }
+        }
+
+        return instance;
+    }
+
+    @Override
+    public void add(Order order) {
+        switch (order.getType()) {
+            case COMPUTE:
+                this.computeOrderStorage.addOrder(order);
+                this.orderTimestampStorage.addOrder(order);
+                break;
+            case NETWORK:
+                this.networkOrderStorage.addOrder(order);
+                this.orderTimestampStorage.addOrder(order);
+                break;
+            case VOLUME:
+                this.volumeOrderStorage.addOrder(order);
+                this.orderTimestampStorage.addOrder(order);
+                break;
+            case ATTACHMENT:
+                this.attachmentOrderStorage.addOrder(order);
+                this.orderTimestampStorage.addOrder(order);
+                break;
+        }
+    }
+
+    @Override
+    public void update(Order order) {
```

Comment:
what if this order was not added to the database?

Reply:
I'm not sure about that, yet.

Reply:
Normally, this should never happen, since update() is only called to an order that has already been activated, and the order is added to the db when it is activated. As I understood, whenever an operation in the db fails, we simply log an error. We will need to develop a tool that reads the log and notifies the site admin of these and other kinds of problems.

Reply:
Yes, currently we are not raising exceptions when operations in the db are performed. We must decide whether logging it's enough.

Reply:
A general comment: I follow fubica's advice of logging the database errors. However, I'm a little bit afraid of keep system operating without the database (I'm trying to devise the sysadmin procedure in case of such failures; as the pain may be high, sometimes is better to limit the damage).

In summary, @llvieira keep the code as it is now (logging the error instead of killing the core). also, keep in mind the admin tool is important, so we could try to develop it as soon as possible

Reply:
Note that the problems with running without stable storage are twofold. Firstly, if there is a crash, I won't be able to recover the system state. This should be rare, and shouldn't be a big issue. Secondly, I won't have accounting information registered. This is potentially more harmful, but should we stop the system to avoid this? I don't think so. I think that the best we can do is to equip the system administrator with tools that will allow her/him to identify these problems as soon as possible, and will fix them accordingly. There are also problems that may happen even if the database is operational. Something may change in the system and a crash may occur before these changes are stored in stable storage. Again, the system administrator needs to come in play to fix possible inconsistencies. I agree that this tool should be develop asap.

Reply:
Agreed, @thiagomanel and @fubica. To stop the whole system in consequence of some database failures really seems a bad idea. In regard to the new tool, I'm willing to help with it.

Reply:
Agreed, @thiagomanel and @fubica. Stopping the whole system in consequence of some database failures really seems a bad idea. In regard to the new tool, I'm willing to help with it.